lib/Sema/CSSimplify.cpp

```diff
@@ -2060,17 +2060,13 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
 
```

Comment:
This dropping shouldn't be needed anymore now that we only look for pattern match element.